appium/webdriver/webdriver.py

```diff
@@ -799,15 +795,24 @@ def pull_folder(self, path):
         }
         return self.execute(Command.PULL_FOLDER, data)['value']
 
-    def push_file(self, path, base64data):
-        """Puts the data, encoded as Base64, in the file specified as `path`.
+    def push_file(self, destination_path, base64data=None, source_path=None):
+        """Puts the data from the file at `source_path`, encoded as Base64, in the file specified as `path`.
 
+        Specify either `base64data` or `source_path`, if both specified default to `source_path`
         :Args:
-         - path - the path on the device
+         - destination_path - the path on the device
          - base64data - data, encoded as Base64, to be written to the file
+         - source_path - path to file to be loaded on device
         """
+        if source_path is None and base64data is None:
+            raise InvalidArgumentException('Must either present base64 data or a source path on local system')
+
+        if source_path is not None:
```

Comment:
You could use `if bool(source_path):`

Reply:
Not sure there's a real benefit to converting to a boolean for the check?

Reply:
I like the current state of the code (with the slightly more informative error message). Thanks.